pkg/cloudprovider/aws/aws.go

```diff
@@ -136,8 +143,10 @@ type Instance struct {
 func (c *CloudProvider) GetInstance(node *v1.Node) (cloudprovider.Instance, error) {
 	var instance *Instance
 
-	id := providerIDToInstanceID(node.Spec.ProviderID)
-
+	id, err := providerIDToInstanceID(node.Spec.ProviderID)
+	if err != nil {
+		return instance, errors.New("no cloud provider id found, it may be set later by cloud controller")
```

Comment:
We're throwing the error away we got from `providerIDToInstanceID` by not using it here. Consider using `errors.Wrap` to wrap it with additional context